example/android/SNetDemo/app/src/main/java/com/example/snetdemo/ImageSegmentationActivity.java

```diff
@@ -131,96 +214,145 @@ protected void onCreate(Bundle savedInstanceState)
 
         textViewProgress = findViewById(R.id.textViewProgress);
         textViewProgress.setVisibility(View.INVISIBLE);
+        textViewProgress.setText("");
 
         textViewResponseTime = findViewById(R.id.textViewResponseTime);
         textViewResponseTime.setText("Service response time (ms):");
         textViewResponseTime.setVisibility(View.INVISIBLE);
 
-
         PackageManager pm = this.getPackageManager();
         if (!pm.hasSystemFeature(PackageManager.FEATURE_CAMERA_ANY))
         {
             isDeviceWithCamera = false;
             btn_GrabCameraImage.setEnabled(false);
         }
 
-        AsyncTask task = new AsyncTask<Object, Object, Object>()
+        disableActivityGUI();
+
+        if( checkAndRequestPermissions() )
         {
+            initApp();
+        }
 
-            protected void onPreExecute()
+    }
+    public boolean checkAndRequestPermissions()
```

Comment:
BTW, why adding permissions to the manifest is not enough?

Reply:
Followed official documentation:
https://developer.android.com/training/permissions/requesting#perm-check